src/iexamine.cpp

```diff
@@ -132,6 +132,61 @@ void iexamine::cvdmachine( player &p, const tripoint & ) {
     p.mod_moves( -1000 );
 }
 
+/**
+ * UI FOR LAB_FINALE NANO FABRICATOR.
+ */
+void iexamine::nanofab( player &p, const tripoint &examp )
+{
+    bool table_exists = false;
+    tripoint spawn_point;
+    for( const auto &valid_location : g->m.points_in_radius( examp, 1, 0 ) ) {
+        if( g->m.ter(valid_location) == t_nanofab_body ) {
+            spawn_point = valid_location;
+            table_exists = true;
+            break;
+        }
+    }
+    if (!table_exists){
+        return;
+    }
+
+    auto nanofab_template= g->inv_map_splice( []( const item &e ) {
+        return ( e.has_var( "NANOFAB_ITEM_ID" ) );
+    }, _( "Introduce Nanofabricator template" ), PICKUP_RANGE, _( "You don't have any usable templates." ) );
+
+    if( !nanofab_template ) {
+        return;
+    }
+
+    //item *it = nanofab_template.get_item();
+    item new_item( nanofab_template->get_var( "NANOFAB_ITEM_ID" ), calendar::turn );
+
+    auto qty = new_item.volume() / units::legacy_volume_factor;
```

Comment:
```suggestion
    auto qty = new_item.volume() / 250_ml;
```
(Because this is not legacy code and your comment above literally says 250 ml.)